tests/helpers/wait-for-loaded-iframe.js

```diff
@@ -33,24 +36,45 @@ export default async function(app, url) {
         iframeWindow.document.addEventListener('DOMContentLoaded', onWindowLoad);
       }
     });
-  });
+  }));
 
+  url = url || "/";
   let times = 0;
 
-  async function tryVisit() {
-    url = url || "/";
+  await run(() => new RSVP.Promise(resolve => {
+    function tryVisit() {
 
-    if (times++ >= 10) {
-      run.cancelTimers();
-    } else if (iframeWindow.visit) {
-      await iframeWindow.visit(url);
-    } else {
-      run.later(async () => await tryVisit(), 10)
+      if (times++ >= 10) {
+        run.cancelTimers();
+        resolve();
+      } else if (iframeWindow.visit) {
+        run(() =>  {
+          iframeWindow.visit(url).then(resolve);
+        });
+      } else {
+        run.later(tryVisit, 10);
+      }
+    }
+    run(tryVisit);
+  }));
+
+  times = 0;
+  await run(() => new RSVP.Promise(resolve => {
+    function waitForContents() {
+
+      if (times++ >= 10) {
+        run.cancelTimers();
+        resolve();
+      } else if (outputContents().length > 0) {
+        resolve();
+      } else {
+        run.later(waitForContents, 10);
+      }
     }
-  }
-  await tryVisit();
+    run(waitForContents);
+  }));
 
   await new RSVP.Promise(function (resolve) {
-    run.later(resolve, 10);
+    run.later(resolve, 1000);
   });
 }
```

Comment:
This was the helper I was unsure about. Thanks!

Reply:
Actually should revert that change.